app/features/notes_generator/tools.py

```diff
@@ -0,0 +1,65 @@
+import os
+import csv
+from typing import Optional
+import docx
+from pdfminer.high_level import extract_text as extract_pdf
+from youtube_transcript_api import YouTubeTranscriptApi
+from docx import Document
+from fpdf import FPDF
+
+def extract_text_from_file(file_path: str) -> str:
+    """extract text based on file type."""
+    if file_path.endswith('.pdf'):
```

Comment:
Use an scalable approach for implementing these conditions. You can check this resource for more info. on that: https://www.linkedin.com/pulse/optimizing-10000-if-else-statements-strategies-best-practices-asim-ammgf/